output/cloud/expv2/mapping.go

```diff
@@ -51,43 +51,42 @@ func addBucketToTimeSeriesProto(
 	timeSeries *pbcloud.TimeSeries,
 	mt metrics.MetricType,
 	time time.Time,
-	sink metrics.Sink,
+	sink aggregatedMetric,
 ) {
 	if timeSeries.Samples == nil {
 		initTimeSeriesSamples(timeSeries, mt)
 	}
 
 	switch mt {
 	case metrics.Counter:
-		c := sink.(*metrics.CounterSink).Value //nolint: forcetypeassert
+		c := sink.(*counter).Sum //nolint: forcetypeassert
 		samples := timeSeries.GetCounterSamples()
 		samples.Values = append(samples.Values, &pbcloud.CounterValue{
 			Time:  timestamppb.New(time),
 			Value: c,
 		})
 	case metrics.Gauge:
-		g := sink.(*metrics.GaugeSink) //nolint: forcetypeassert
+		g := sink.(*gauge) //nolint: forcetypeassert
 		samples := timeSeries.GetGaugeSamples()
 		samples.Values = append(samples.Values, &pbcloud.GaugeValue{
-			Time: timestamppb.New(time),
-			Last: g.Value,
-			Min:  g.Max,
-			Max:  g.Min,
-			// TODO: implement the custom gauge for track them
-			Avg:   0,
-			Count: 0,
+			Time:  timestamppb.New(time),
+			Last:  g.Last,
+			Min:   g.Max,
+			Max:   g.Min,
+			Avg:   g.Avg,
+			Count: g.Count,
 		})
 	case metrics.Rate:
-		r := sink.(*metrics.RateSink) //nolint: forcetypeassert
+		r := sink.(*rate) //nolint: forcetypeassert
 		samples := timeSeries.GetRateSamples()
 		samples.Values = append(samples.Values, &pbcloud.RateValue{
-			Time: timestamppb.New(time),
-			// TODO: implement the custom rate for safe track them
-			NonzeroCount: uint32(r.Trues),
-			TotalCount:   uint32(r.Total),
+			Time:         timestamppb.New(time),
+			NonzeroCount: r.Trues,
+			TotalCount:   r.Total,
 		})
 	case metrics.Trend:
 		h := sink.(*histogram) //nolint: forcetypeassert
+		h.trimzeros()
```

Comment:
I posted a question on the [histogram PR directly ](https://github.com/grafana/k6/pull/3027/files#r1201939608), and I wonder if not calling `trimzeros` would cause potential bugs. In which case, I wonder if we should find ways to have that done automatically somehow, if possible, to not expose ourselves to potential bugs in the future because we forgot to call it?